doc/tut3.rst

```diff
@@ -0,0 +1,277 @@
+=======================
+Nim Tutorial (Part III)
+=======================
+
+:Author: Arne Döring
+:Version: |nimversion|
+
+.. contents::
+
+
+Introduction
+============
+
+  "With Great Power Comes Great Responsibility." -- Spider Man
+
+This document is a tutorial for the macros of the *Nim* programming
+language.  A macro enables to formulate and distribute Nim syntax tree
+transformations as a normal library.  The arguments of a macro are
+passed a syntax tree, and its job is it to create another syntax tree
+for the compiler.  The way this works in Nim is, whenever the compiler
+encounters a call expression to a macro, The compiler evaluates the
+macro at compile time with the syntax tree from the invocation, and
+then it replacing the call to the macro by the result of the macro.
+
+Examples of things that can be implemented in macros:
+
+ * An assert macro that prints both sides of a comparison operator, if
+the assertion fails. ``myAssert(a == b)`` that converts to
+``if a != b: quit($a " != " $b)``
+
+ * A debug macro that prints the value and the name of the symbol.
+``myDebugEcho(a)`` that converts to ``echo "a: ", a``
+
+ * Symbolic differenciation of an expression.
+``diff(a*pow(x,3) + b*pow(x,2) + c*x + d, x)``  that converts to
+``3*a*pow(x,2) + 2*a*x + c``
+
+Macro Arguments
+---------------
+
+There are two ways to pass arguments to a macro, either typed or
```

Comment:
The first sentence doesn't have a connection with the rest of the paragraph, I would put it at the end, as a short intro for the next two sections.